connector/servicegraphconnector/connector_test.go

```diff
@@ -675,14 +674,19 @@ func TestVirtualNodeServerLabels(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NoError(t, conn.ConsumeTraces(context.Background(), td))
 
-	conn.store.Expire()
-	assert.Eventually(t, func() bool {
-		return conn.store.Len() == 0
-	}, 100*time.Millisecond, 2*time.Millisecond)
-	require.NoError(t, conn.Shutdown(context.Background()))
+	// Wait for metrics to be generated with timeout
+	deadline := time.Now().Add(5 * time.Second)
+	var metrics []pmetric.Metrics
+	for time.Now().Before(deadline) {
```

Comment:
Side note: you can keep using the `assert.Eventually` or `assert.EventuallyWithT`, the main advantage being the error message in case of failure.

Reply:
Re-added again